elements/tags.lua

```diff
@@ -621,39 +652,42 @@ local function getTagFunc(tagstr)
 		for bracket in tagstr:gmatch(_PATTERN) do
 			local tagFunc = funcPool[bracket] or tags[bracket:sub(2, -2)]
 			if(not tagFunc) then
-				local tagName, tagStart, tagEnd = getTagName(bracket)
-
+				local tagName, prefixEnd, suffixStart, suffixEnd, customArgs, outputLength = getBracketData(bracket)
 				local tag = tags[tagName]
 				if(tag) then
-					tagStart = tagStart - 2
-					tagEnd = tagEnd + 2
-
-					if(tagStart ~= 0 and tagEnd ~= 0) then
-						local prefix = bracket:sub(2, tagStart)
-						local suffix = bracket:sub(tagEnd, -2)
+					if(2 - prefixEnd ~= 1 and suffixStart - suffixEnd ~= 1) then
+						local prefix = bracket:sub(2, prefixEnd)
+						local suffix = bracket:sub(suffixStart, suffixEnd)
 
 						tagFunc = function(unit, realUnit)
-							local str = tag(unit, realUnit)
-							if(str) then
-								return prefix .. str .. suffix
+							local str = tag(unit, realUnit, string.split(',', customArgs))
+							if(str and str ~= '') then
+								return prefix .. (outputLength and utf8shorten(str, outputLength) or str) .. suffix
 							end
 						end
-					elseif(tagStart ~= 0) then
-						local prefix = bracket:sub(2, tagStart)
+					elseif(2 - prefixEnd ~= 1) then
+						local prefix = bracket:sub(2, prefixEnd)
 
 						tagFunc = function(unit, realUnit)
-							local str = tag(unit, realUnit)
-							if(str) then
-								return prefix .. str
+							local str = tag(unit, realUnit, string.split(',', customArgs))
+							if(str and str ~= '') then
+								return prefix .. (outputLength and utf8shorten(str, outputLength) or str)
 							end
 						end
-					elseif(tagEnd ~= 0) then
-						local suffix = bracket:sub(tagEnd, -2)
+					elseif(suffixStart - suffixEnd ~= 1) then
+						local suffix = bracket:sub(suffixStart, -2)
 
 						tagFunc = function(unit, realUnit)
-							local str = tag(unit, realUnit)
-							if(str) then
-								return str .. suffix
+							local str = tag(unit, realUnit, string.split(',', customArgs))
```

Comment:
I just didn't want to have an extra if in there.
```lua
customArgs and string.split(',', customArgs)
```
would result in only the 1st return being passed, while the rest would be ignored, but if you really want I could change it to
```lua
local str
if(customArgs) then
  tag(unit, realUnit, string.split(',', customArgs))
else
  str = tag(unit, realUnit)
end
```
Also, we usually use functions like `table.insert` or in this case `string.split` as is.

Reply:
I find it better with the if statement yes.  However it won't be enough, because `getBracketData` returns `tag:match('%((.-)%)') or ''` for customArgs.

I am not sure how big of an impact `strplit` vs `string.split` has, it is probably tiny, so I don't insist on that one.

Reply:
It returns `tag:match('%((.-)%)') or ''` to avoid any errors from since `customArgs and string.split(',', customArgs)` doesn't work and of that if statement is too ugly 🙈 If we choose to use ifs, which I don't want to do 😿 , I'll just remove that `or ''`.

Reply:
I just want to avoid the added cost of calling `string.split` on an empty string on tags that don't use the args. Unless I miss something, we do [this](https://github.com/oUF-wow/oUF/blob/1d24857231e5ff62ed793c73401d1cae37e70e28/elements/tags.lua#L690-L758) just to save on calling `table.unpack` where possible at the cost of using 68 SLOCs for something we could do in 16. We should probably test the perf benefits from that one. Either way.